lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
+  attr_reader :name, :address, :phone,
+              :services, :registered_vehicles, :collected_fees
+              :administer_written_test
+  def initialize(name:, address:, phone:)
     @name = name
     @address = address
     @phone = phone
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @administer_written_test
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless services.include?("Vehicle Registration")
```

Comment:
Great use of a guard clause!!!  The other way you could write this (some people don't like the unless) `return nil if !services.include?("Vehicle Registration")`. Either way is fine.